tsc/frr/sonic_frr_update_process.md

```diff
@@ -0,0 +1,78 @@
+-   Responsibility of SONiC FRR release maintainer
+	-   Default 12 months assignment (or until the next upgrade)
+	-   Upgrade FRR version in Nov release, resolve SONiC FRR upgrade integration issues
+  -   If there is a need for May release upgrade due to feature requirement or other reasons, it should be requested in the May release plan and have the same maintainer to handle the second upgrade
+	-   Triage and fix SONiC FRR issues when applicable. Fix may come from SONiC contributors or from FRR community, maintainer is responsible to drive the fix to unblock SONiC community
+	-   Submit fixes to FRR project, submit new FRR topo test to FRR project if there is a gap
+	-   Release maintainer to subscribe to FRR project, and be the FRR Point-of-Contact on behalf of SONiC
+	-   Bring in FRR vulnerabilities and critical patches to SONiC
+
+# SONiC FRR vulnerability and patch upgrade in between SONiC releases
+
+-   FRR CVE Fixes
+	-   Reference nvd.nist.gov cvss v3.x [rating](https://nvd.nist.gov/vuln-metrics/cvss#)
+	-   Bring in Critical and High FRR CVE patches into SONiC
+	-   Sample of [Critical](https://nvd.nist.gov/vuln/search/results?form_type=Advanced&results_type=overview&search_type=all&isCpeNameSearch=false&cpe_vendor=cpe%3A%2F%3Afrrouting&cpe_product=cpe%3A%2F%3A%3Afrrouting&cvss_version=3&cvss_v3_severity=CRITICAL), sample of [High](https://nvd.nist.gov/vuln/search/results?form_type=Advanced&results_type=overview&search_type=all&isCpeNameSearch=false&cpe_vendor=cpe%3A%2F%3Afrrouting&cpe_product=cpe%3A%2F%3A%3Afrrouting&cvss_version=3&cvss_v3_severity=HIGH)
+	-   There are regular security scan in Azure pipeline, CVEs will be filed to sonic-buildimage/issues
+	-   SONiC FRR release maintainer should subscribe to nvd.nist.gov for FRR alerts
+	-   Need a process to bring in CVEs to earlier SONiC releases too (open to suggestions)
+
+-   Patch FRR Bug Fixes
+	-   SONiC FRR release maintainer should subscribe to FRR project to bring in critical patch which is applicable to SONiC
+
+# SONiC FRR Upgrade Steps
+-   Create sonic-frr branch for the target FRR version
+	-   Contact release manager
+-   Find new package dependencies
+	  -   Upload newly required packages to a common location (azure?)
+-   Submodule update to new FRR commit id.
+-   Code changes
+    -   Version change in Makefiles
+    -   New Makefiles for new packages (if any.)
+    -   Port the patches
+	    -   Evaluate whether existing FRR patches still applicable to new FRR version
+		 -   Apply the old patches into new FRR version, and generate new patch files
+			 -   Keep original credentials
+	    -   If the changes are already present in new FRR version, discard the old patch file
+		-   If the patch does not apply, manually merge the changes and resolve any conflicts
+    -   Build and verify
+	    -   Use PTF on local server, – or –
+	    -   Manually verify BGP, VRF, IPv4, IPv6 (on sonic-vs.)
+    -   Create PR
```

Comment:
I would recommend the new PR details clearly like what are the patch changes like, new patches, removed patches, modified patches, etc. This would help in review.
I have defined a template in FRR 8.5.1 PR. May be we can use it? https://github.com/sonic-net/sonic-buildimage/pull/15965

Reply:
Will update the proposal with your template